File: AbstractFactory/java/FemaleHumanFactory.java
public class FemaleHumanFactory implements HumanFactory{
    public Human createBlackHuman(){
        return new FemaleBlackHuman();
    }

    public Human createWhiteHuman(){
        return new FemaleWhiteHuman();
    }

    public Human createYellowHuman(){
        return new FemaleYellowHuman();
    }
}

File: AbstractFactory/C++/Source/FemaleHumanFactory.cpp
#include "../Include/FemaleHumanFactory.hpp"
#include "../Include/FemaleBlackHuman.hpp"
#include "../Include/FemaleWhiteHuman.hpp"
#include "../Include/FemaleYellowHuman.hpp"

Human* FemaleHumanFactory::createBlackHuman()
{
    return dynamic_cast<Human*>(new FemaleBlackHuman());
}

Human* FemaleHumanFactory::createWhiteHuman()
{
    return dynamic_cast<Human*>(new FemaleWhiteHuman());
}

Human* FemaleHumanFactory::createYellowHuman()
{
    return dynamic_cast<Human*>(new FemaleYellowHuman());
}